chain/types/src/utils.rs

```diff
@@ -41,10 +42,58 @@ sol!(
     }
 );
 
+sol!(
+    #![sol(abi)]
+    #![sol(rpc)]
+    function multiSend(bytes memory transactions) public payable;
+);
+
 lazy_static::lazy_static! {
     static ref MINTER_ROLE_VALUE: primitives::FixedBytes<32> = keccak256("MINTER_ROLE");
 }
 
+/// Struct to make a multisend transaction, mainly used by safe instances
+#[derive(Debug, Clone)]
+pub struct MultisendCallOnlyTransaction {
+    // data paylaod encoded with selector
+    pub encoded_data: Bytes,
+    // target address
+    pub to: alloy::primitives::Address,
+    // payable eth sending along the tx
+    pub value: U256,
+}
+
+/// Methods for Multisend transaction
+impl MultisendCallOnlyTransaction {
+    /// encode a multisend transaction
+    pub fn encode_packed(&self) -> Vec<u8> {
+        let tx_operation_bytes: [u8; 1] = hex!("00"); // call only
+
+        let value = (
+            tx_operation_bytes,                  // 1 bytes
+            self.to,                             // 20 bytes
+            U256::from(self.value),              // 32 bytes
+            U256::from(self.encoded_data.len()), // 32 bytes
+            self.encoded_data.clone(),           // bytes
+        );
+        value.abi_encode_packed()
```

Comment:
With a little bit of tweaking, this could be implementable as `AsRef<&[u8]>`

Reply:
Eliminating the vector allocation completely.

Reply:
You mean `AsRef<[u8]>` :)

Reply:
Yes.